### ISSUE TITLE
Supply

### ISSUE THREAD

**Issue by 457Stonehouse** [birgersp/dcs-autogft#75]
Enhancement thought at a conceptual level - Add a pool of supply units (air, sea land) and allow a limited number to be active (simulating a sides quality of logistics and distribution control ie more active = better logistics). The pool size and active number can vary according to coalition. Possibly set up at the coalition level a limited pool of fuel for vehicles and ammo for vehicles which is decremented each time a resupply group is sent out. Possibly link actual ground targets to this pool so that on destruction of the ground targets the pool is reduced. The ground targets might be also tied into a strategic supply generation rate. eg all supply ground targets intact = pool of fuel and ammo refreshed at interval X mins. If 80% of supply ground targets intact then pool reduced by a factor due to loss of supply structures and also pool is refreshed from outside theatre X% slower. Also may reduce the rate of regeneration of supply units (if allowed at all) and number of active supply groups - simulating disruption to logistics command and control.

When a combat group reaches a predefined supply level (low fuel and/or ammo) have them hold and defend their current location and request resupply. This would spawn/activate a group of supply units whose size depends on the requester group size. The resupply group would then be given a route to the requester group which could resupply on the supply groups arrival. Once resupplied the combat group is free to act as per normal within the scripts definitions. The supply group would then need to return to base in game before despawning/ going inactive and freeing up one of the limited number of resupply slots. If destroyed on the way to the rendezvous with the requester group then the requester group will need to generate a new request which will be filled when an active resupply slot is available and there is supply units available. If destroyed on the way back to the rear area then the pool of supply units is reduced.  In either case destruction of the resupply group means the resupply slot is freed up. If the requester group is destroyed then the supply groups should be retasked to RTB and if it arrives safely then the supplies would go back into the pool of fuel and ammo. If it is possible to allocate a portion of the total required resupply to each unit in the resupply group then it would be possible for to cater for loss of some portion of the supply group to mean a partial resupply for the combat group. This might mean they can go back to normal ops for a limited time before requesting resupply again or indeed mean they don't get over the threshold and need to request resupply again. Which would have to wait for logistics to become available. Possibly may want to assign different resupply priorities to different composition combat groups (eg HQ has determined that armour should be a priority over infantry). Might open resupply by helo for post Korean war scenarios. Possibly if combat group is out of supply for a given time it will try to retreat to a resupply location and cancel the resupply request. Possibly allow forward points to become designated resupply starting points as the forward line of troops advances having the effect of shortening supply lines. Probably this would be based on a time value eg. location captured at time A, can become a resupply start point if not cut off from the rear area and has been held for time A plus X mins. Also should allow for naval resupply missions to forward supply points or combat groups within a given radius of the coast. Possibly strategic supply refreshment could be via naval convoy or strategic airlift. 

Possibly could expand the system to cover air bases and assigned air units down the track. eg low avgas would = request for resupply and reduction of air missions generated out of that base. Definitely a future item though as it is outside the ground aspect of this script.

This would definitely make a change to how the scenario plays out as the units at the pointy end would be the ones that run out of fuel and ammo first making the advance stall if resupply is not forthcoming. It also means that logistic groups either air, land or sea and supply locations become useful and legitimate targets which have an immediate impact on combat group performance.

**Comment by birgersp** [comment]
Great suggestion! I love the idea that logistic groups are legitimate targets to halt the progress of the enemy task forces.

I/we are going to have to try and break it down some, into more specific/detailed goals and (ultimately) issues.

Firsty, I need to consider if it is feasible to expand the current task force definitions to implement these features, or if it should be a separate type. This definition seems bigger than a task force, possibly a team (group) of task forces that recieves their orders from a common "master" (You're welcome to suggest type definition names).

**Comment by 457Stonehouse** [comment]
Yeah I was thinking the logistics control needs to be at a coalition level and some sort of logistics command module will monitor queued resup requests and assign resup missions either on a FIFO or priority basis. Could start with FIFO and move to priority sorted requests later if it makes it simpler to get it going, especially when it is still going to be FIFO within a priority anyway (ie the oldest priority 1 is served first and before priority 2s. When a priority 2 gets a go it is the oldest request.) 

Probably best place to start is to try to identify conceptual functions and try to come up with a process flow design overall to act as a guide to build it. There are some fairly obvious functions to start with and likely could act as a proof of concept (not saying any of it is easy)

- monitoring group supply levels to trigger resupply requests and retask a group needing resupply (halt and defend) and exclude it from "normal" autogft tasking. Also needs to pick up when a group is resupplied and available for normal autogft tasking.
- monitoring resupply requests and generating the resupply mission when supplies and resupply slots are available. 
-updating supply pool and available supply mission slots. eg in simplest form supply mission leaves HQ then pool is reduced and so is available resupply slots. If a supply group is destroyed or arrives home the available resupply slots are increased. When slots = 0 or supplies = 0 no resupply missions can be generated. 

Will try to think about it some more if I get some spare time (something which is a bit sparse at present). Going to have to revisit some fine detail in your script as it's grown faster than I've kept up and while I get the gist (I hope) when you say task force (group of groups?) it's likely I'm missing something.
